app/concepts/todo/create.rb

```diff
@@ -10,41 +12,46 @@ class Todo::Create < Trailblazer::Operation
   #         create a default `todo_list`
   #         associate `todo_list` with `user`
   #         associate `todo` with `todo_list`
+
+  contract do
+    property :title, validates: { presence: true }
+    property :description
+    property :todo_list_id
+  end
+
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
-
-    # build the todo
-    todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
-
-    # find or create the parent list for the todo
-    if params[:todo_list_id].blank?
-      todo_list = TodoList.find_or_create_by(name: "Default To-do List")
-
-      # Find or create the owner of the todo_list:
-      ##### If current_user_id is present, fetch that user
-      ##### otherwise, create a new one
-      current_user =
-        if params[:current_user] == nil
-          if params[:current_user_id] == nil
-            User.create!(fullname: "Guest")
-          else
-            User.find(params[:current_user_id])
-          end
-        else
-          params[:current_user]
-        end
+    model.list = get_list(params)
 
-      # assign the todo_list to the user
-      todo_list.user = current_user
-      todo_list.save!
-    else # i.e. if params[:todo_list_id] is not nil
-      todo_list = TodoList.find(params[:todo_list_id])
+    # validate and save 
+    validate(params) do
+      contract.save
     end
+  end
```

Comment:
It's good to see that you've separated the concern of creating a `todo_list` and a `user` from the main concern of creating a `todo`.